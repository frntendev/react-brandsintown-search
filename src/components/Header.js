import React from "react";
import { NavLink } from "react-router-dom";
import { css } from "react-emotion";

const styles = {
  navigation: css`
    width: 100%;
    text-align: right;
    display: flex;
    align-items: center;
    background: #6b6b6b0d;
    ul {
      flex: 1;
    }
    label: navigation;
  `,
  listItem: css`
    list-style: none;
    display: inline-block;
    height: 4.28em;
    line-height: 4.28em;
    list-style: none;
    background: transparent;
    transition: all 0.3s ease-in-out;
    cursor: pointer;
    border: none;
    overflow: hidden;
    text-align: center;
    label: listItem;
  `,
  icon: css`
    display: none !important;
  `,
  link: css`
    display: inline-block;
    text-decoration: none;
    padding: 0 20px;
    background: #bfbfbf;
    color: #fff;
    margin-right: 10px;
    min-width: 70px;
    font-weight: bold;
    font-size: 1rem;
    transition: all 0.3s ease;
    label: link;
    &:hover {
      background: #3fa99c;
    }
  `,
  activeLink: css`
    background: #49c5b6;
    &:hover {
      background: #3fa99c;
    }
  `,
  logo: css`
    flex: 1;
    text-align: left;
    font-size: 2.3rem;
    font-weight: 800;
    text-decoration: underline;
    font-style: italic;
    padding-left: 10px;
    label: logo;
    @media (max-width: 480px) {
      display: none;
    }
  `
};

const Header = () => {
  return (
    <header>
      <nav className={styles.navigation}>
        <span className={styles.logo}>Brandsintown search</span>
        <ul>
          <li className={styles.listItem}>
            <NavLink
              to="/"
              exact
              className={styles.link}
              activeClassName={styles.activeLink}
            >
              <span>Search</span>
            </NavLink>
          </li>
          <li className={styles.listItem}>
            <NavLink
              to="/favorite"
              className={styles.link}
              activeClassName={styles.activeLink}
            >
              <span>Favorites</span>
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
