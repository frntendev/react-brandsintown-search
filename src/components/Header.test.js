import React from "react";
import { shallow, mount, configure } from "enzyme";
import Header from "./Header";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("Header", () => {
  test("renders component itself correctly", () => {
    const renderedComponent = shallow(<Header />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
