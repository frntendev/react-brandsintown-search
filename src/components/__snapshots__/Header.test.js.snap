// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders component itself correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="css-1excjay-navigation"
      >
        <span
          className="css-eirv9e-logo"
        >
          Brandsintown search
        </span>
        <ul>
          <li
            className="css-10t90bd-listItem"
          >
            <NavLink
              activeClassName="css-thlpwr"
              ariaCurrent="true"
              className="css-xr45i9-link"
              exact={true}
              to="/"
            >
              <span>
                Search
              </span>
            </NavLink>
          </li>
          <li
            className="css-10t90bd-listItem"
          >
            <NavLink
              activeClassName="css-thlpwr"
              ariaCurrent="true"
              className="css-xr45i9-link"
              to="/favorite"
            >
              <span>
                Favorites
              </span>
            </NavLink>
          </li>
        </ul>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="css-eirv9e-logo"
          >
            Brandsintown search
          </span>,
          <ul>
            <li
              className="css-10t90bd-listItem"
            >
              <NavLink
                activeClassName="css-thlpwr"
                ariaCurrent="true"
                className="css-xr45i9-link"
                exact={true}
                to="/"
              >
                <span>
                  Search
                </span>
              </NavLink>
            </li>
            <li
              className="css-10t90bd-listItem"
            >
              <NavLink
                activeClassName="css-thlpwr"
                ariaCurrent="true"
                className="css-xr45i9-link"
                to="/favorite"
              >
                <span>
                  Favorites
                </span>
              </NavLink>
            </li>
          </ul>,
        ],
        "className": "css-1excjay-navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Brandsintown search",
            "className": "css-eirv9e-logo",
          },
          "ref": null,
          "rendered": "Brandsintown search",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="css-10t90bd-listItem"
              >
                <NavLink
                  activeClassName="css-thlpwr"
                  ariaCurrent="true"
                  className="css-xr45i9-link"
                  exact={true}
                  to="/"
                >
                  <span>
                    Search
                  </span>
                </NavLink>
              </li>,
              <li
                className="css-10t90bd-listItem"
              >
                <NavLink
                  activeClassName="css-thlpwr"
                  ariaCurrent="true"
                  className="css-xr45i9-link"
                  to="/favorite"
                >
                  <span>
                    Favorites
                  </span>
                </NavLink>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="css-thlpwr"
                  ariaCurrent="true"
                  className="css-xr45i9-link"
                  exact={true}
                  to="/"
                >
                  <span>
                    Search
                  </span>
                </NavLink>,
                "className": "css-10t90bd-listItem",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "css-thlpwr",
                  "ariaCurrent": "true",
                  "children": <span>
                    Search
                  </span>,
                  "className": "css-xr45i9-link",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Search",
                  },
                  "ref": null,
                  "rendered": "Search",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="css-thlpwr"
                  ariaCurrent="true"
                  className="css-xr45i9-link"
                  to="/favorite"
                >
                  <span>
                    Favorites
                  </span>
                </NavLink>,
                "className": "css-10t90bd-listItem",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "css-thlpwr",
                  "ariaCurrent": "true",
                  "children": <span>
                    Favorites
                  </span>,
                  "className": "css-xr45i9-link",
                  "to": "/favorite",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Favorites",
                  },
                  "ref": null,
                  "rendered": "Favorites",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="css-1excjay-navigation"
        >
          <span
            className="css-eirv9e-logo"
          >
            Brandsintown search
          </span>
          <ul>
            <li
              className="css-10t90bd-listItem"
            >
              <NavLink
                activeClassName="css-thlpwr"
                ariaCurrent="true"
                className="css-xr45i9-link"
                exact={true}
                to="/"
              >
                <span>
                  Search
                </span>
              </NavLink>
            </li>
            <li
              className="css-10t90bd-listItem"
            >
              <NavLink
                activeClassName="css-thlpwr"
                ariaCurrent="true"
                className="css-xr45i9-link"
                to="/favorite"
              >
                <span>
                  Favorites
                </span>
              </NavLink>
            </li>
          </ul>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="css-eirv9e-logo"
            >
              Brandsintown search
            </span>,
            <ul>
              <li
                className="css-10t90bd-listItem"
              >
                <NavLink
                  activeClassName="css-thlpwr"
                  ariaCurrent="true"
                  className="css-xr45i9-link"
                  exact={true}
                  to="/"
                >
                  <span>
                    Search
                  </span>
                </NavLink>
              </li>
              <li
                className="css-10t90bd-listItem"
              >
                <NavLink
                  activeClassName="css-thlpwr"
                  ariaCurrent="true"
                  className="css-xr45i9-link"
                  to="/favorite"
                >
                  <span>
                    Favorites
                  </span>
                </NavLink>
              </li>
            </ul>,
          ],
          "className": "css-1excjay-navigation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Brandsintown search",
              "className": "css-eirv9e-logo",
            },
            "ref": null,
            "rendered": "Brandsintown search",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="css-10t90bd-listItem"
                >
                  <NavLink
                    activeClassName="css-thlpwr"
                    ariaCurrent="true"
                    className="css-xr45i9-link"
                    exact={true}
                    to="/"
                  >
                    <span>
                      Search
                    </span>
                  </NavLink>
                </li>,
                <li
                  className="css-10t90bd-listItem"
                >
                  <NavLink
                    activeClassName="css-thlpwr"
                    ariaCurrent="true"
                    className="css-xr45i9-link"
                    to="/favorite"
                  >
                    <span>
                      Favorites
                    </span>
                  </NavLink>
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="css-thlpwr"
                    ariaCurrent="true"
                    className="css-xr45i9-link"
                    exact={true}
                    to="/"
                  >
                    <span>
                      Search
                    </span>
                  </NavLink>,
                  "className": "css-10t90bd-listItem",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "css-thlpwr",
                    "ariaCurrent": "true",
                    "children": <span>
                      Search
                    </span>,
                    "className": "css-xr45i9-link",
                    "exact": true,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Search",
                    },
                    "ref": null,
                    "rendered": "Search",
                    "type": "span",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="css-thlpwr"
                    ariaCurrent="true"
                    className="css-xr45i9-link"
                    to="/favorite"
                  >
                    <span>
                      Favorites
                    </span>
                  </NavLink>,
                  "className": "css-10t90bd-listItem",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "css-thlpwr",
                    "ariaCurrent": "true",
                    "children": <span>
                      Favorites
                    </span>,
                    "className": "css-xr45i9-link",
                    "to": "/favorite",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Favorites",
                    },
                    "ref": null,
                    "rendered": "Favorites",
                    "type": "span",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
